### DISCUSSION TITLE
Bookplayer: Zipped MP3 Files Issue

### DISCUSSION THREAD

**Discussion by gandalftheking** [TortugaPower/BookPlayer#841] [General]
Apart from not being able to natively display book covers ;-).... My issue is described by the image attached - I have tried exporting as a zip file via itunes including a cheeky jpg book cover, and then a zip file with just the mp3 files on their own - same error each time. M4B's behave very well, but MP3's not so much.

Any ideas to cure this please?

![IMG_9987](https://user-images.githubusercontent.com/114857805/194302216-264fbe9b-9399-4861-908e-82d438c32c46.PNG)

**Comment by GianniCarlo** [comment]
Hi @gandalftheking , not sure how to export zip files from iTunes, could you share one of these zip files for testing?

**Reply by gandalftheking** [reply]
No I mean importing zip into itunes as recommended way of dealing with mp3's. Once in bookplayer i then get the message as posted....

**Reply by GianniCarlo** [reply]
ooooh ok, could you tell me what's the structure of the zip file?
I've tested importing a zip file with this structure and can't reproduce the issue
folder
- file1.mp3
- file2.mp3
- nested
  - file3.mp3
  -  file4.mp3

There's one extra thing I have to mention, file names or folder names can't have the character `/` (e.g. my book part 1/10), because the '/' character is a special character to describe folder/file paths (e.g. folder/nested/file4.mp3)

**Comment by aandnota** [comment]
Does the zip import only work with mp3 and m4b? I’ve tried doing it with a zip of .flac, and it never works.

**Reply by GianniCarlo** [reply]
@aandnota  do you mean that it imports and then it doesn't play the .flac file? or that it doesn't even show the file after finishing importing the zip?

**Reply by aandnota** [reply]
I mean that it doesn’t even complete the import. Nothing happens after I send the zip to BookPlayer.

**Reply by GianniCarlo** [reply]
🤔 if not even the import screen is being shown after sending it, it may be a problem with one of the event listeners when there's a new file (of any type) to import. After sending the zip, and BookPlayer not showing the import screen, can you force close it from the app switcher, and open BookPlayer again? The listeners will be created again and do a sweep of pending files to import and the import screen should show up listing them. Question, which method are you using for importing? I may have to look into it and figure out when it's erroring out

**Reply by aandnota** [reply]
Initially I was using the following import method: Files > Share > BookPlayer. That method would open up BookPlayer, but wouldn't show me the import dialog. 

After your suggestion I tried importing from BookPlayer and that worked.

**Comment by gandalftheking** [comment]
OP here - my issue was using RAR files not ZIP. Bookplayer only likes Zip. Now able to import multiple mp3's and a cover image and it all works ok. Would still like bigger images in the library view though ;-)
